CorePayments/src/main/java/com/paypal/android/corepayments/graphql/GraphQLClient.kt

```diff
@@ -36,13 +32,18 @@ class GraphQLClient internal constructor(
         "Origin" to coreConfig.environment.graphQLEndpoint
     )
 
-    suspend fun send(graphQLRequestBody: JSONObject, queryName: String? = null): GraphQLResult {
+    suspend fun send(
+        graphQLRequestBody: JSONObject,
+        queryName: String? = null,
+        headers: Map<String, String>? = null
+    ): GraphQLResult {
         val body = graphQLRequestBody.toString()
         val urlString = if (queryName != null) "$graphQLURL?$queryName" else graphQLURL
         val httpRequest = HttpRequest(URL(urlString), HttpMethod.POST, body, httpRequestHeaders)
+        headers?.forEach { (key, value) -> httpRequestHeaders.put(key, value) }
```

Comment:
```suggestion
        headers?.forEach { (key, value) -> httpRequestHeaders.put(key, value) }
        val httpRequest = HttpRequest(URL(urlString), HttpMethod.POST, body, httpRequestHeaders)
```
I feel like it makes sense to add headers before constructing the `HttpRequest`. This will prevent bugs in the future if the `HttpRequest` constructor one day makes its own copy of `httpRequestHeaders`.